### PR TITLE
HOWL-14: Modify controllers and routes

### PR DESCRIPTION
**Pull request by KJMoon**
[HOWL-14](https://www.notion.so/HOWL-14-Modify-controllers-and-routes-to-include-users-and-auth-60418e866a9e4f88b096b91d4f44b998?pvs=4)

This modifies the routes and controllers in the backend to include a middleware function for Auth0.
The function ensures that the user is authenticated.

### CODE REVIEW

File: backend/controllers/main.js [hunk 1]
```diff
@@ -15,10 +34,13 @@ module.exports = {
...
     },
 
+    // Get the howl from the user
     getHowl: async (req, res, next) => {
         try {
-            const { rows } = await pool.query(`SELECT * FROM howls WHERE id = ${req.params.id}`);
+            const postID = req.params.id;
             
+            const { rows } = await pool.query(`SELECT * FROM howls WHERE id = ${postID}`);
+
```
Comment (yuzinu) [inline backend/controllers/main.js]: Let's keep our naming consistent 🙂 
`howlId` should be what this is called

File: backend/controllers/main.js [hunk 2]
```diff
@@ -30,16 +52,21 @@ module.exports = {
...
     },
 
+    // Create a new howl for the user
     createHowl: async (req, res, next) => {
         try {
-            await pool.query('INSERT INTO howls (caption) VALUES($1)', [req.body.caption]);
+            const caption = req.body.caption;
+            // const username = req.oidc.user.nickname;
+
+            await pool.query(`INSERT INTO howls (caption, howler_id) SELECT '${caption}', id FROM users WHERE username = 'yuzinu'`);
```
Comment (yuzinu) [inline backend/controllers/main.js]: This part is still in progress right? Looks like we're not doing the username yet

File: backend/middleware/auth.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    ensureAuth: function (req, res, next) {
+        if (req.oidc.isAuthenticated()) {
+            return next();
+        } else {
+            res.redirect("/");
+        }
+    },
+};
```
Comment (yuzinu) [inline backend/middleware/auth.js:9]: This is *chef's kiss* 😘🤏

**Review by yuzinu — APPROVED** [review]
Great job with this!

File: backend/controllers/main.js (hunk 1 above)
Comment (KJMoon) [reply backend/controllers/main.js]: Changed naming in commit [f91027b](https://github.com/yuzinu/howler/pull/13/commits/f91027bcf9edc1dd735885f90cf400b9199bd0d9) 😉

File: backend/controllers/main.js (hunk 2 above)
Comment (KJMoon) [reply backend/controllers/main.js]: Added username into the query in commit [f91027b](https://github.com/yuzinu/howler/pull/13/commits/f91027bcf9edc1dd735885f90cf400b9199bd0d9)

File: backend/middleware/auth.js (hunk 3 above)
Comment (KJMoon) [reply backend/middleware/auth.js:9]: TYVM 😍

**Review by yuzinu — APPROVED** [review]

**Pull request merged**